src/Sensor/MemoryMeasurement.java

```diff
@@ -7,18 +7,22 @@
 public class MemoryMeasurement extends Measurement {
     
     MemoryMeasurement(Sigar _sigar){
-        super("Memory measure");
-        sigar = _sigar;
+        super("Memory measure", _sigar);
     }
 
     public String getActualMeasure(){
-        Long value = (long) -1;
+        Long memoryUsageValueBytes = (long) -1;
```

Comment:
You return only **memoryUsageValueMegabytes**, so this variable shouldn't be declared here, but in place of definition

Use -1L, don't mix convertion with literals